core/src/main/java/org/apache/spark/metrics/prometheus/client/exporter/TextFormatWithTimestamp.java

```diff
@@ -0,0 +1,178 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.spark.metrics.prometheus.client.exporter;
+
+import java.io.IOException;
+import java.io.Writer;
+import java.util.Enumeration;
+
+import io.prometheus.client.Collector;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+public class TextFormatWithTimestamp {
+    private static final Logger logger = LoggerFactory.getLogger(TextFormatWithTimestamp.class);
+
+    /**
+     * Content-type for text version 0.0.4.
+     */
+    public static final String CONTENT_TYPE_004 = "text/plain; version=0.0.4; charset=utf-8";
+
+    private static StringBuilder jsonMessageLogBuilder = new StringBuilder();
```

Comment:
Usage of this variable is questionable for a couple of reasons:
- it just keeps growing, it's never cleared or re-initialized. As a consequence from the second call of write it will have invalid content + it acts as a memory leak.
- its usage pattern (`writer.write(blah);appendToJsonMessageLogBuilder("blah")`) is pretty verbose, it should be factored out. 
- it's not thread safe (and it's not documented)
- I don't think accessing it as a static member everywhere is a good design. It should either
  - be passed around as method parameter
  - or changed to an instance method. The static write004 could instantiate a new `TextFormatWithTimestamp` and call write on that.

Reply:
This class has been refactored.